streams/src/main/java/org/apache/kafka/streams/kstream/internals/KeyValueStoreMaterializer.java

```diff
@@ -46,16 +47,9 @@ public KeyValueStoreMaterializer(final MaterializedInternal<K, V, KeyValueStore<
     public StoreBuilder<?> materialize() {
         KeyValueBytesStoreSupplier supplier = (KeyValueBytesStoreSupplier) materialized.storeSupplier();
         if (supplier == null) {
-            switch (materialized.storeType()) {
-                case IN_MEMORY:
-                    supplier = Stores.inMemoryKeyValueStore(materialized.storeName());
-                    break;
-                case ROCKS_DB:
-                    supplier = Stores.persistentTimestampedKeyValueStore(materialized.storeName());
-                    break;
-                default:
-                    throw new IllegalStateException("Unknown store type: " + materialized.storeType());
-            }
+            supplier = materialized.storeSuppliers.keyValueStore(
```

Comment:
nit: since we simplified this part of the code so much, can we go one step further and refactor so that the `supplier` variable is final?